File: Solutions/Easy/3507. Minimum Pair Removal to Sort Array I/Solution.cs
public class Solution {
    public int MinimumPairRemoval(int[] nums) {
        int n = nums.Length, res = 0;
        if(n == 1) return 0;
        List<int> numsList = new List<int>(nums);
        while(!IsNonDecreasing(numsList))
        {
            int minimumSum = Int32.MaxValue, pos = -1;
            for(int i = 0; i < numsList.Count - 1; i++)
            {
                if(minimumSum > numsList[i] + numsList[i+1])
                {
                    pos = i;
                    minimumSum = numsList[i] + numsList[i + 1];
                }
            }
            numsList[pos] = minimumSum;
            numsList.RemoveAt(pos + 1);
            res++;
        }
        return res;
    }

    private bool IsNonDecreasing(List<int> numsList)
    {
        for(int i = 0; i < numsList.Count - 1; i++)
        {
            if(numsList[i] > numsList[i + 1]) return false;
        }
        return true;
    }
}

File: Solutions/Easy/3507. Minimum Pair Removal to Sort Array I/Solution.py
class Solution:
    def minimumPairRemoval(self, nums: List[int]) -> int:
        n: int = len(nums)
        result: int = 0
        while(not(self.nonDecreasing(nums))):
            result += 1
            minSum, pos = (5000000, -1)
            for i in range(0, n - 1):
                if(minSum > nums[i] + nums[i + 1]):
                    minSum = nums[i] + nums[i + 1]
                    pos = i
            n -= 1
            nums.pop(pos + 1)
            nums[pos] = minSum
        return result
    
    def nonDecreasing(self, nums: List[int]) -> bool:
        n: int = len(nums)
        for i in range(1, n):
            if(nums[i - 1] > nums[i]):
                return False
        return True